test/util/print.test.js

```diff
@@ -1,20 +1,48 @@
 const math = require('mathjs');
 
 const print = require('../../lib/util/print');
+const Node = require('../../lib/node');
 
 const TestUtil = require('../TestUtil');
 
-function testPrint(exprStr, outputStr) {
+// to create nodes, for testing
+const opNode = Node.Creator.operator;
+const constNode = Node.Creator.constant;
+const symbolNode = Node.Creator.symbol;
+
+function testPrintStr(exprStr, outputStr) {
   const input = math.parse(exprStr);
   TestUtil.testFunctionOutput(print, input, outputStr);
 }
 
+function testPrintNode(node, outputStr) {
+  TestUtil.testFunctionOutput(print, node, outputStr);
+}
+
 describe('print asciimath', function () {
   const tests = [
     ['2+3+4', '2 + 3 + 4'],
     ['2 + (4 - x) + - 4', '2 + (4 - x) - 4'],
     ['2/3 x^2', '2/3 x^2'],
     ['-2/3', '-2/3'],
   ];
-  tests.forEach(t => testPrint(t[0], t[1]));
+  tests.forEach(t => testPrintStr(t[0], t[1]));
 });
+
+describe('print with parenthesis', function () {
```

Comment:
perfect! exactly what I was thinking

this rarely shows up in the tests so it's fair that you didn't think of it

(fun fact: all the tests looked like this for a bit, until I realized it'd be much easier to parse strings :p)